src/Entities/Player.cpp

```diff
@@ -1673,35 +1673,36 @@ void cPlayer::SetIP(const AString & a_IP)
 
 unsigned int cPlayer::AwardAchievement(const eStatistic a_Ach)
 {
-	eStatistic Prerequisite = cStatInfo::GetPrerequisite(a_Ach);
-
+	// Old 1.8 - 1.11 processing
+	AString nameNew = allStatisticsNames[a_Ach];
+	AString Prerequisite = cStatManager::GetPrerequisite(nameNew);
 	// Check if the prerequisites are met
-	if (Prerequisite != statInvalid)
+	if (Prerequisite != "")
 	{
 		if (m_Stats.GetValue(Prerequisite) == 0)
 		{
 			return 0;
 		}
 	}
 
-	StatValue Old = m_Stats.GetValue(a_Ach);
+	StatValue Old = m_Stats.GetValue(nameNew);
```

Comment:
I changed this because the name lookup has already been done and the GetValue with the enum value (a_Ach) simply performs another lookup